ptftplib/dhcpserver.py

```diff
@@ -171,6 +171,9 @@ def __init__(self, pkt):
 
         # Strip off the ethernet frame and check the IP packet type. It should
         # be UDP (0x11)
+        # for python3 it's bytes, need to decode to str first
+        if type(pkt) is bytes:
+            pkt = pkt.decode(errors='ignore')
```

Comment:
My point is that I think this should happen in `DHCPServer.serve_forever()` (line 251):

```
import six
...
pkt = DhcpPacket(six.ensure_str(data))
self.handle_dhcp_request(pkt)
```

Reply:
how about this:
https://github.com/zhouronghua/ptftpd/pull/1#issue-617601375

Reply:
Yes. Can you get this into this PR? Don't forget to add `six` to the `requirements.txt` file.

Reply:
> Yes. Can you get this into this PR? Don't forget to add `six` to the `requirements.txt` file.

sorry, it doesn't work, the content of the data have changed, i prefer to drop this PR, and adapt ord only